sympy/matrices/matrices.py

```diff
@@ -384,15 +384,55 @@ def cofactor_matrix(self, method="berkowitz"):
                          lambda i, j: self.cofactor(i, j, method))
 
     def det(self, method="bareiss", iszerofunc=None):
-        """Computes the determinant of a matrix.  If the matrix
-        is at most 3x3, a hard-coded formula is used.
-        Otherwise, the determinant using the method `method`.
+        """Computes the determinant of a matrix.
 
+        Parameters
+        ==========
+
+        method : string, optional
+            Specifies the algorithm used for computing the matrix determinant.
+
+            If the matrix is at most 3x3, a hard-coded formula is used and the
+            specified method is ignored. Otherwise, it defaults to ``bareiss``.
+
+            If it is set to ``bareiss``, Bareiss' fraction-free algorithm will
+            be used.
+
+            If it is set to ``berkowitz``, Berkowitz' algorithm would be used.
+
+            Otherwise, if it is set to ``lu``, LU decomposition would be used.
+
+            .. note::
+                For backward compatibility, legacy keys like **bareis** ,
+                **det_lu** can still be used to indicate corresponding
+                methods.
+                And the keys are also **case-insensitive** for now.
+                However, we would discourage you to use the keys in the manner.
+
+        iszerofunc : FunctionType or None, optional
+            If it is set to ``None``, it will be defaulted to ``_iszero`` if the
+            method is set to ``bareiss``, and ``_is_zero_after_expand_mul`` if
```

Comment:
The usage of keys could possibly pointed out as strings. ("bareiss" and similar)